Don't force C++ standard - boost-math-1.90 requires C++14
Use fortran linker for correct rpath to gcc fortran libraries

--- src/CMakeLists.txt~	2020-01-02 12:15:39.000000000 +0100
+++ src/CMakeLists.txt	2020-09-14 11:50:40.474978389 +0200
@@ -24,6 +24,7 @@
 endif()
 
 add_library(${PROJECT_NAME} SHARED ${SOURCES} ${HEADERS})
+target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
 target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include>)
 set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR} VERSION ${PROJECT_VERSION})
 set_target_properties(${PROJECT_NAME} PROPERTIES COMPILE_FLAGS "-DEIGEN_QUADPROG_EXPORT")
@@ -61,6 +62,8 @@
 
 if(${USE_F2C})
   target_link_libraries(${PROJECT_NAME} libf2c.a)
+elseif(NOT ${USE_FORTRAN_SUBDIRECTORY})
+  set_property(TARGET ${PROJECT_NAME} PROPERTY LINKER_LANGUAGE Fortran)
 endif()
 
 install(
